### DISCUSSION TITLE
GL.iNet GL-MT6000 running OpenWrt with Crusader 0.2

### DISCUSSION THREAD

**Discussion by richb-hanover** [Zoxc/crusader#63] [Wi-Fi routers and access points]
## Test conditions

* Firmware: OpenWrt 23.05.4 running on GL.iNet GL-MT6000 router
* Crusader Client GUI client: 0.2 running on macOS 12.7.5 Macbook
* Crusader Server: 0.2 pre-built for aarch64 running on router
* Crusader Server was installed by downloading the pre-built aarch64...musl...Linux binary

## Observations

* Ethernet looks really good. Consistently near 1gbps, very low and consistent latency
* Wi-Fi 2.4GHz - not sure why upload is faster. Latency well under control
* Wi-Fi 5GHz - Latency is higher, especially for remote test, but still well in control

## Ethernet
![plot 2024-08-31 09 04 22](https://github.com/user-attachments/assets/7b92dfdc-dcdb-41e8-a3d5-8d5c05b58b1d)

## Wi-Fi 2.4GHz 3ft (1m) - 20MHz channel, RSSI -41dB, 802.11n
![plot 2024-08-31 13 58 12](https://github.com/user-attachments/assets/4f044a62-8f73-48d6-ae1e-847e0391b1a0)
![plot 2024-08-31 13 57 15](https://github.com/user-attachments/assets/94d499d9-ddef-42ca-8bf8-b210109313fb)
![plot 2024-08-31 13 56 18](https://github.com/user-attachments/assets/0ed3b42f-e0c2-45af-81b7-f9bcd2e3d24b)
![plot 2024-08-31 13 55 07](https://github.com/user-attachments/assets/b037f612-8e5e-4b0a-8631-8fb0636a87b1)

## Wi-Fi 5GHz 3 ft (1m) - 80MHz channel, RSSI -53dB, 802.11ac 
![plot 2024-08-31 14 10 00](https://github.com/user-attachments/assets/81ee0fdc-bd56-4e84-8bb7-bd894706454a)
![plot 2024-08-31 14 09 11](https://github.com/user-attachments/assets/9aaa252b-e6e2-496f-8aac-45913a9562ea)
![plot 2024-08-31 14 08 22](https://github.com/user-attachments/assets/e22561e9-9a0a-4bc3-a809-177d6856bb38)

## Wi-Fi 2.4GHz 25ft (8m) - 20MHz channel, RSSI -65dB, 802.11n
![plot 2024-08-31 14 14 58](https://github.com/user-attachments/assets/87093761-4d73-419c-935f-36d0a77e0446)
![plot 2024-08-31 14 14 11](https://github.com/user-attachments/assets/4d5aab2c-8802-44fe-9f89-fcc7213c79ce)
![plot 2024-08-31 14 13 25](https://github.com/user-attachments/assets/e0662650-ea8b-4734-90e5-5b22eb40764b)

## Wi-Fi 5GHz 25 ft (8m) - 80MHz channel, RSSI -79dB, 802.11ac
![plot 2024-08-31 14 19 00](https://github.com/user-attachments/assets/bd20735a-d62a-4162-9a24-651722fabe85)
![plot 2024-08-31 14 18 12](https://github.com/user-attachments/assets/916f5d94-7677-42f8-b6de-d491d13ee044)
![plot 2024-08-31 14 16 46](https://github.com/user-attachments/assets/116c1ac6-4e42-4a5d-b00b-2a7a48def2bd)

**Comment by richb-hanover** [comment]
**Update: Added more tests using a Crusader server on an RPi4 on Ethernet (instead of running it on the router). The Macbook Pro running the Crusader GUI was on WiFi**

## Wi-Fi 2.4GHz 3ft (1m) to RPi4 on Ethernet - 8 streams

![plot 2024-09-08 15 40 24](https://github.com/user-attachments/assets/de8a9633-2005-4d18-9e69-20bb0e8cd4ca)

## Wi-Fi 2.4GHz 3ft (1m) to RPi4 on Ethernet - 1 stream

![plot 2024-09-08 15 41 28](https://github.com/user-attachments/assets/e11d1fd6-208b-4144-9a72-24af149388ce)

## Wi-Fi 5GHz 3ft (1m) to RPi4 on Ethernet - 8 streams
![plot 2024-09-08 15 45 35](https://github.com/user-attachments/assets/43997e7a-eb89-4e2a-a558-43a14178ac45)
![plot 2024-09-08 15 44 45](https://github.com/user-attachments/assets/c5012f93-cb5e-4d3f-96d2-f7477b7db3f9)
![plot 2024-09-08 15 43 38](https://github.com/user-attachments/assets/9eb368a3-dcf6-4b0a-8e66-a76e18b86745)

## Wi-Fi 5GHz 3ft (1m) to RPi4 on Ethernet - 1 stream
![plot 2024-09-08 15 46 33](https://github.com/user-attachments/assets/ee2020ab-94d2-4f86-9673-a819441f6ec9)

**Comment by Zoxc** [comment]
I'm curious how wired WAN <-> LAN routing performance with Cortex A53 scales. You'd use a single stream in Crusader for that. No flow offloading or SQM in OpenWrt. It would be nice to have some benchmarks at 0.8 Ghz, 1 Ghz, 1.3 Ghz, 1.5 Ghz, 1.8 Ghz and 2 Ghz (fixed clocks). These are values which align with various SoCs. Testing with `piece_of_cake` also with speeds set to 1 Gbps would be nice too.

You'd want some faster computers running Crusader for that. My Raspberry Pi 4 doesn't handle symmetric gigabit, not sure about 5.

A test of the Beryl AX with stock and a fixed 1.3 Ghz clock would be interesting too to see if it lines up.
